heareval/predictions/task_predictions.py

```diff
@@ -257,22 +304,43 @@ def label_vocab_nlabels(embedding_path: Path) -> Tuple[pd.DataFrame, int]:
     return (label_vocab, nlabels)
 
 
+def dataloader_from_split_name(
+    embedding_path: Path, nlabels: int, split_name: str, batch_size: int = 64
+) -> DataLoader:
+    print(f"Getting embeddings for split: {split_name}")
+
+    return DataLoader(
+        SplitMemmapDataset(embedding_path, nlabels, split_name),
+        batch_size=batch_size,
+        # We don't shuffle because it's slow.
+        # Also we want predicted labels in the same order as
+        # target labels.
+        shuffle=False,
+    )
+
+
 def task_predictions_train(
     embedding_path: Path, scene_embedding_size: int, timestamp_embedding_size: int
 ) -> torch.nn.Module:
     metadata = json.load(embedding_path.joinpath("task_metadata.json").open())
     label_vocal, nlabels = label_vocab_nlabels(embedding_path)
 
     if metadata["embedding_type"] == "scene":
-        predictor = RandomProjectionPrediction(
+        predictor = PredictionModel(
             scene_embedding_size, nlabels, metadata["prediction_type"]
         )
     elif metadata["embedding_type"] == "event":
-        predictor = RandomProjectionPrediction(
+        predictor = PredictionModel(
             timestamp_embedding_size, nlabels, metadata["prediction_type"]
         )
     else:
         raise ValueError(f"embedding_type {metadata['embedding_type']} unknown.")
+    # train on CPU
+    # TODO: FIXME
```

Comment:
What's broken?

Reply:
just that we want to specify more parameters here, like device